app/uk/gov/hmrc/cardpaymentfrontend/models/extendedorigins/ExtendedBtaSa.scala

```diff
@@ -42,4 +44,82 @@ class ExtendedBtaSa extends ExtendedOrigin {
   override def surveyReturnMessageKey: String = "payments-survey.bta.return-message"
   override def surveyIsWelshSupported: Boolean = true
   override def surveyBannerTitle: String = serviceNameMessageKey
+
+  def checkYourAnswersRows(request: JourneyRequest[AnyContent])(implicit messages: Messages): Seq[CheckYourAnswersRow] = {
+    val referenceRow = CheckYourAnswersRow(
+      "btasa.reference.title",
+      Seq(reference(request).dropRight(1)), //Do not display the final K in the Utr in the CYA table),
+      None
+    )
+
+    val dateRow = CheckYourAnswersRow(
+      "btasa.date.title",
+      Seq(Messages("btasa.date.today")),
+      Some(Link(
+        Call("GET", "this/that"),
+        "btasa.date-change-link",
+        "btasa.date.change-link.text"
+      ))
+    )
+
+    val amountRow = CheckYourAnswersRow(
+      "btasa.amount.title",
+      Seq(amount(request)),
+      Some(Link(
+        Call("GET", "this/that"),
+        "pfvat-reference-change-link",
+        "pfvat.reference.change-link.text"
+      ))
+    )
+
+    val addressRow = {
+      val maybeAddress: Option[Address] = request.readFromSession[Address](request.journeyId, Keys.address)
+      CheckYourAnswersRow(
+        "btasa.address.title",
+        maybeAddress match {
+          case Some(addr) => Seq(addr.line1, addr.line2.getOrElse(""), addr.city.getOrElse(""), addr.county.getOrElse(""), addr.postcode, addr.country).filter(_.nonEmpty)
+          case None       => Seq.empty
+        },
+        Some(Link(
+          Call("GET", "this/that"),
+          "ptasa-address-change-link",
```

Comment:
Should this not be BtaSa

Reply:
Eventually, working up the links is in the next ticket